### ISSUE TITLE
Version conflict for doctrine/orm

### ISSUE THREAD

**Issue by cnkt** [composer/composer#6712]
My `composer.json`:

```json
{
  "require": {
    "silex/silex": "~2.0",
    "doctrine/orm": "^2.5",
    "twig/twig": "~1.0",
    "symfony/console": "^3.0",
    "cakephp/collection": "dev-master",
    "knplabs/knp-menu": "^2.1@dev",
    "zendframework/zendframework1": "^1.12@dev",
    "phpoffice/phpexcel": "dev-master",
    "dflydev/doctrine-orm-service-provider": "^2.0",
    "ircmaxell/random-lib": "1.1.*@dev",
    "gedmo/doctrine-extensions": "2.4.*@dev",
    "jbroadway/urlify": "1.0.*@dev",
    "igorw/get-in": "^1.0@dev",
    "symfony/security": "^3.1",
    "symfony/security-bundle": "^3.1",
    "symfony/twig-bridge": "^3.0",
    "symfony/var-dumper": "^3.1",
    "symfony/debug-bundle": "^3.1",
    "symfony/translation": "^3.1",
    "sorien/silex-dbal-profiler": "~2.0",
    "sinasalek/silex-pagerfanta-provider": "1.0.*@dev",
    "fzaninotto/faker": "^1.6@dev",
    "kriswallsmith/assetic": "^1.3@dev",
    "tedivm/stash": "0.14.*",
    "league/flysystem": "^1.1@dev",
    "vlucas/phpdotenv": "^2.1@dev",
    "anahkiasen/underscore-php": "dev-master",
    "mouf/html.widgets.statsgrid": "~1.0",
    "codeguy/upload": "dev-master",
    "imagine/imagine": "~0.5.0",
    "leongkui/silex-php-redis": "*",
    "phpmailer/phpmailer": "dev-master",
    "swiftmailer/swiftmailer": "^5.4",
    "guzzlehttp/guzzle": "^6.1",
    "illuminate/encryption": "^5.1",
    "phpunit/php-timer": "~1.0",
    "elasticsearch/elasticsearch": "~2.0@beta",
    "mk-j/php_xlsxwriter": "dev-dev",
    "fabpot/goutte": "^3.1",
    "cakephp/chronos": "@stable",
    "beberlei/DoctrineExtensions": "^1.0",
    "akeneo-labs/spreadsheet-parser": "^1.2",
    "php-amqplib/php-amqplib": "2.6.*",
    "sendgrid/sendgrid": "~5.0.4",
    "aura/payload": "^3.0",
    "box/spout": "^2.5",
    "sentry/sentry": "^1.1",
    "datadog/php-datadogstatsd": "0.3.*",
    "friendsofphp/php-cs-fixer": "^1.12",
    "ramsey/uuid-doctrine": "^1.2",
    "sabre/xml": "^2.0",
    "zendframework/zendxml": "^1.0",
    "saxulum/saxulum-doctrine-orm-manager-registry-provider": "^3.1",
    "saxulum/saxulum-console": "~3.1",
    "cboden/ratchet": "0.3.*",
    "react/zmq": "0.2.*|0.3.*",
    "mongodb/mongodb": "^1.2@dev",
    "saxulum/saxulum-doctrine-orm-commands": "dev-master",
    "sorien/silex-pimple-dumper": "~2.0",
    "predis/service-provider": "^2.0@dev",
    "symfony/cache": "^4.0@dev"
  },
  "include-path": [],
  "autoload": {
    "psr-0": {
      "O2": "src/"
    },
    "classmap": ["lib/"]
  },
  "minimum-stability": "dev",
  "require-dev": {
    "sstalle/php7cc": "^1.0",
    "phpmd/phpmd": "dev-master",
    "escapestudios/symfony2-coding-standard": "~2.0"
  },
  "repositories": [
    {
      "type": "vcs",
      "url": "https://github.com/AlexStef/saxulum-doctrine-orm-commands"
    }
  ]
}

```

Output of `composer diagnose`:

```
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
require.cakephp/collection : unbound version constraints (dev-master) should be avoided
require.phpoffice/phpexcel : unbound version constraints (dev-master) should be avoided
require.anahkiasen/underscore-php : unbound version constraints (dev-master) should be avoided
require.codeguy/upload : unbound version constraints (dev-master) should be avoided
require.leongkui/silex-php-redis : unbound version constraints (*) should be avoided
require.phpmailer/phpmailer : unbound version constraints (dev-master) should be avoided
require.cakephp/chronos : unbound version constraints (@stable) should be avoided
require.saxulum/saxulum-doctrine-orm-commands : unbound version constraints (dev-master) should be avoided
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
composer update --prefer-dist
```

I get the following output:

```
   1/1:	http://packagist.org/p/provider-latest$7b3de6e55cac870a58f9e92d03478ea4b0e8016b8c0c5fb0c6de1cd5efbb4a1c.json
    Finished: success: 1, skipped: 0, failure: 0, total: 1
Loading composer repositories with package information
Updating dependencies (including require-dev)


  [Composer\DependencyResolver\SolverProblemsException]
    Problem 1
      - Installation request for doctrine/orm 2.5.11 -> satisfiable by doctrine/orm[v2.5.11].
      - doctrine/orm v2.5.11 requires doctrine/common >=2.5-dev,<2.9-dev -> no matching package found.
    Problem 2
      - doctrine/orm v2.5.11 requires doctrine/common >=2.5-dev,<2.9-dev -> no matching package found.
      - ramsey/uuid-doctrine 1.4.1 requires doctrine/orm ^2.5 -> satisfiable by doctrine/orm[v2.5.11].
      - Installation request for ramsey/uuid-doctrine ^1.2 -> satisfiable by ramsey/uuid-doctrine[1.4.1].
  Potential causes:
   - A typo in the package name
   - The package is not available in a stable-enough version according to your minimum-stability setting
     see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more details.
  Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.


update [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [--with-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>]...
```

And I expected this to happen:

Update the dependencies to required versions.

Additional info:

Doctrine team says the problem is on composer side: https://github.com/doctrine/doctrine2/issues/6738

**Comment by akomm** [comment]
I had similar issue with doctrine/inflector not being seen by composer. 
I cleared the cached packages - issue remained
I removed the lock file - issue remained

This fixed it:
I set manually the third party dependencies in the root composer.json untill all errors were gone.
then I reverted the composer.json, remove lock file, deleted vendor.

After that composer install worked correctly.
I did clone doctrine/orm in another folder and composer install worked right ahead, without manually including dependencies.

I think composer stored something incorrectly inside its local settings in .composer. And it were not the cached packages.
